File: learn-java/src/main/java/boki/learnjava/flux/FluxEx3.java
package boki.learnjava.flux;

import lombok.extern.slf4j.Slf4j;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.Optional;

@Slf4j
public class FluxEx3 {
    // Mono + Mono = Flux
    public static void main(String[] args) {
        // Flux<Object> flux = Mono.justOrEmpty(null)
        Flux<Object> flux = Mono.justOrEmpty(Optional.empty())
            .concatWith(Mono.justOrEmpty("Jobs"));

        flux.subscribe(data -> log.info("# result: {}", data));
    }
}

File: learn-kt/src/main/kotlin/boki/learn/flux/FluxEx3.kt
package boki.learn.flux

import reactor.core.publisher.Mono

fun main() {
    // Mono + Mono = Flux
    val flux = Mono.justOrEmpty<Any>(null)
        // 람다로 처리하면 값이 출력되지 않는다(적절한 인자 필요)
        // .concatWith { Mono.justOrEmpty<Any>("Jobs") }
        .concatWith(Mono.justOrEmpty<Any>("Jobs"))

    flux.subscribe { println("# result: $it") }
}